hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/federation/access/TriplesResponse.java

```diff
@@ -7,21 +7,24 @@
 
 public interface TriplesResponse extends DataRetrievalResponse<Iterable<Triple>>
 {
-	Iterable<Triple> getTriples();
-
 	/**
-	 * Returns the number of triples that are returned by the {@link #getTriples()}. 
+	 * Returns the number of triples that are returned by the {@link #getResponseData()}.
 	 */
-	default int getSize(){
-		final Iterable<Triple> triples = getTriples();
-		if (triples instanceof Collection) {
-			return ((Collection<Triple>) triples).size();
-		}
-		// Fallback to manual count
-		int count = 0;
-		for ( Iterator<Triple> it = triples.iterator(); it.hasNext(); it.next() ){
-			count++;
+	default int getSize() {
+		try {
+			final Iterable<Triple> triples = getResponseData();
+			if ( triples instanceof Collection ) {
+				return ((Collection<Triple>) triples).size();
+			}
+			// Fallback to manual count
+			int count = 0;
+			for ( Iterator<Triple> it = triples.iterator(); it.hasNext(); it.next() ) {
+				count++;
+			}
+			return count;
+		} catch ( UnsupportedOperationDueToRetrievalError e ) {
+			// intentionally do nothing
 		}
-		return count;
+		return -1;
```

Comment:
I think it is better to throw that same kind of exception in this case.

That's a general comment which you can apply for all the getSize() functions here.